Python/bytecodes.c

```diff
@@ -966,6 +975,9 @@ dummy_func(
             STAT_INC(STORE_ATTR, deferred);
             _PyAttrCache *cache = (_PyAttrCache *)next_instr;
             DECREMENT_ADAPTIVE_COUNTER(cache->counter);
+            #else
+            assert(counter >= 0); /* prevent compiler warning */
```

Comment:
I think this will still warn on debug builds. Maybe something like this (also in the other spots):

```suggestion
            (void)counter;  // Unused.
```